frontend/src/selectors/mineSelectors.js

```diff
@@ -33,3 +33,27 @@ export const getCurrentMineTypes = createSelector(
     }
   }
 );
+
+export const getTransformedMineTypes = createSelector(
+  [getMines, getMineGuid],
+  (mines, mineGuid) => {
+    const mine_types = {
+      mine_tenure_type_code: [],
+      mine_commodity_code: [],
+      mine_disturbance_code: [],
+    };
+    if (mineGuid) {
+      mines[mineGuid].mine_type.map((type) => {
```

Comment:
So, a couple thoughts on this..
If you're not returning anything then you'll be better off using `.forEach()` instead of `.map()`

But I also think we can clean this up quite a bit.. something like (haven't tested this):
```javascript
(mines, mineGuid) => {
  const extract = (key) => type.mine_type_detail.filter((detail) => detail[key]).map((detail) => detail[key]);
  return {
    mine_tenure_type_code: mineGuid ? mines[mineGuid].mine_type.(({ mine_tenure_type code }) => mine_tenure_type_code),
    mine_commodity_code: extract("mine_commodity_code"),
    mine_disturbance_code: extract("mine_disturbance_code"),
  };
}
```

Reply:
```
export const getTransformedMineTypes = createSelector(
  [getMines, getMineGuid],
  (mines, mineGuid) => {
    const mine_types = {
      mine_tenure_type_code: [],
      mine_commodity_code: [],
      mine_disturbance_code: [],
    };
    if (mineGuid) {
      mines[mineGuid].mine_type.forEach((type) => {
        mine_types.mine_tenure_type_code.push(type.mine_tenure_type_code);
        const getDetails = (key) =>
          type.mine_type_detail.forEach((detail) => {
            if (detail[key]) mine_types[key].push(detail[key]);
          });
        getDetails("mine_commodity_code");
        getDetails("mine_disturbance_code");
      });
      return mine_types;
    }
  }
);
```

@NWCalvank I updated it to this, but I think this is less clear. thoughts?

Reply:
also, how do you get your code blocks to be in colour??

Reply:
Ahh.. I see why you're doing that way. I didn't spot that it was using `type` lexically...

1) If you don't think my solution is any clearer then you don't need to use it.
2) But if you do, you can just pass `type` as an argument to the `getDetails` function.

Something like:
```javascript
(mines, mineGuid) => {
  const getDetails = (key, type) => type.mine_type_detail.filter((detail) => detail[key]).map((detail) => detail[key]);
  const types = mineGuid ? mines[mineGuid].mine_type : [];
  return {
    mine_tenure_type_code: types.map(({ mine_tenure_type code }) => mine_tenure_type_code),
    mine_commodity_code: types.map((type) => getDetails("mine_commodity_code", type)),
    mine_disturbance_code: types.map((type) => getDetails("mine_disturbance_code", type)),
  };
}
```
3) Putting the language name after the backticks gives you highlighting :)